docs/azure/dependency-injection.md

```diff
@@ -0,0 +1,233 @@
+
+Say you have two storage accounts – one for private information and one for public information. Your application transfers data from the public to private storage account after some operation. You need to have two storage service clients. To set this up in `Startup.ConfigureServices`:
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    builder.AddBlobServiceClient(Configuration.GetSection("PublicStorage"));
+    builder.AddBlobServiceClient(Configuration.GetSection("PrivateStorage"))
+      .WithName("PrivateStorage");
+  });
+}
+```
+
+In your controllers, you can access the named service clients using the `IAzureClientFactory`:
+
+```csharp
+public class HomeControllers : Controller
+{
+  private BlobServiceClient publicStorage, privateStorage;
+
+  public HomeController(BlobServiceClient defaultClient, IAzureClientFactory<BlobServiceClient> clientFactory)
+  {
+    this.publicStorage = defaultClient;
+    this.privateStorage = clientFactory.GetClient("PrivateStorage");
+  }
+}
+```
+
+The un-named service client is still available in the same way as before. Named clients are additive to this.
+
+## Configure a new retry policy
+
+At some point, you might want to change the default settings for a service client. You may want different retry settings or to use a different service API version, for example. You can set the retry settings globally or on a per service basis. Say you have the following `appsettings.json` file:
+
+```json
+{
+  "AzureDefaults": {
+    "Retry": {
+      "maxTries": 3
+    }
+  },
+  "KeyVault": {
+    "VaultUri": "https://mykeyvault.vault.azure.net"
+  },
+  "Storage": {
+    "ServiceUri": "https://store1.storage.windows.net"
+  },
+  "CustomStorage": {
+    "ServiceUri": "https://store2.storage.windows.net"
+  }
+  ```
+
+  You could change the retry policy depending on your needs like so:
+  
+  ```csharp
+  public void ConfigureServices(IServiceCollection services)
+{
+  services.AddAzureClients(builder =>
+  {
+    // Establish the global defaults
+    builder.ConfigureDefaults(Configuration.GetSection("AzureDefaults"));
+    builder.UseCredential(new DefaultAzureCredential());
+
+    // A Key Vault Secrets client using the global defaults
+    builder.AddSecretClient(Configuration.GetSection("KeyVault"));
+
+    // A Storage client with a custom retry policy
+    builder.AddBlobServiceClient(Configuration.GetSection("Storage"))
+      .ConfigureOptions(options => options.Retry.MaxRetries = 10);
+
+    // A named storage client with a different custom retry policy
+    builder.AddBlobServiceClient(Configuration.GetSection("CustomStorage"))
+      .WithName("CustomStorage")
+      .ConfigureOptions(options => {
+        options.Retry.Mode = Azure.Core.RetryMode.Exponential;
+        options.Retry.MaxRetries = 5;
+        options.Retry.MaxDelay = TimeSpan.FromSections(120);
+      });
+  });
+}
+```
```

Comment:
```suggestion
```csharp
public void ConfigureServices(IServiceCollection services)
{
    services.AddAzureClients(builder =>
    {
        // Establish the global defaults
        builder.ConfigureDefaults(Configuration.GetSection("AzureDefaults"));
        builder.UseCredential(new DefaultAzureCredential());
    
        // A Key Vault Secrets client using the global defaults
        builder.AddSecretClient(Configuration.GetSection("KeyVault"));
    
        // A Storage client with a custom retry policy
        builder.AddBlobServiceClient(Configuration.GetSection("Storage"))
            .ConfigureOptions(options => options.Retry.MaxRetries = 10);
    
        // A named storage client with a different custom retry policy
        builder.AddBlobServiceClient(Configuration.GetSection("CustomStorage"))
            .WithName("CustomStorage")
            .ConfigureOptions(options =>
            {
                options.Retry.Mode = Azure.Core.RetryMode.Exponential;
                options.Retry.MaxRetries = 5;
                options.Retry.MaxDelay = TimeSpan.FromSections(120);
            });
    });
}
```
```